siocli/lib/Siocli.cpp

```diff
@@ -272,41 +284,71 @@ int commandPrompt(const ClientParameters& params)
                         }
                     }
 
-                    boost::trim(line);
-                    if (line.empty()) {
-                        if (params.m_stdinIsTerminal && !params.m_useReadline)
-                            std::cout << prompt << std::flush;
-                        continue;
+                    // Detect string value state
+                    lineStartsInStringValue = lineEndsInStringValue;
+                    bool isEscaped = false;
+                    for (char c : line) {
+                        if (c == '\'' && !isEscaped) {
+                            isInStringValue = !isInStringValue;
+                        }
+                        isEscaped = c == '\\';
+                        lineEndsInStringValue = isInStringValue;
+                        if (params.m_printDebugMessages) {
+                            std::cout << "debug: char: " << c << " | isEscaped:" << isEscaped
```

Comment:
This one tends to produce too much output when `m_printDebugMessages` is true, i.e. it will print for every (!!!) character - that's too much. `m_printDebugMessages` was not originally meant to have that much output. I recommend to change this into the following:
```cpp
#if 0
// your debug output, i.e. std::cout << ....
#endif
```
and change `#if 0` into `#if 1` temporarily when you really need to debug that, but keep `#if 0` committed into repo.